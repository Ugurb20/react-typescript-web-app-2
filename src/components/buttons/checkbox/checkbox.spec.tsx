import Checkbox from "@components/buttons/checkbox/checkbox";
import {fireEvent, render, screen} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";


describe('Checkbox', () => {
  it('should render', () => {
    const { container } = render(<Checkbox checked={false} />);
    expect(container).toMatchSnapshot();
  });

  it('should render with label', () => {
    const label = "Label Includes Random Characters";
    // Check if label variable is visible in the snapshot

    const { container } = render(<Checkbox checked={false} label={label} />);

    expect(container).toMatchSnapshot();
    const labelElement = screen.getByText(label);
    expect(labelElement).toBeInTheDocument();
  });

  it('should render with checked', () => {
    const checked = true;
    // Check if checked variable is visible in the snapshot

    const { container } = render(<Checkbox checked={checked} />);

    expect(container).toMatchSnapshot();
  });

  it('should render with onChecked', () => {
    const onChecked = jest.fn();
    // Check if onChecked variable is visible in the snapshot

    const { container } = render(<Checkbox checked={false} onChecked={onChecked} />);

    expect(container).toMatchSnapshot();
  });

  it('should render with label, checked and onChecked', () => {
    const label = "Label Includes Random Characters";
    const checked = true;
    const onChecked = jest.fn();
    // Check if label, checked and onChecked variables are visible in the snapshot

    const { container } = render(<Checkbox checked={checked} label={label} onChecked={onChecked} />);

    expect(container).toMatchSnapshot();
    const labelElement = screen.getByText(label);
    expect(labelElement).toBeInTheDocument();
  });

  it('onChecked should be fired when clicked.', () => {
    const onChecked = jest.fn();
    // Check if onChecked function is called when clicked

    const { getByTestId } = render(<Checkbox checked={false} onChecked={onChecked} />);
    const checkbox = getByTestId("checkbox");

    checkbox.click();
    expect(onChecked).toBeCalled();
  });

});
