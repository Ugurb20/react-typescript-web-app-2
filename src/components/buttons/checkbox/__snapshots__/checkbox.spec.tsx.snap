// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render 1`] = `
<div>
  <div
    class="checkbox"
    data-testid="checkbox"
  >
    <label />
    <input
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Checkbox should render with checked 1`] = `
<div>
  <div
    class="checkbox__checked"
    data-testid="checkbox"
  >
    <label />
    <input
      checked=""
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="DoneIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
      />
    </svg>
  </div>
</div>
`;

exports[`Checkbox should render with label 1`] = `
<div>
  <div
    class="checkbox"
    data-testid="checkbox"
  >
    <label>
      Label Includes Random Characters
    </label>
    <input
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Checkbox should render with label, checked and onChecked 1`] = `
<div>
  <div
    class="checkbox__checked"
    data-testid="checkbox"
  >
    <label>
      Label Includes Random Characters
    </label>
    <input
      checked=""
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="DoneIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
      />
    </svg>
  </div>
</div>
`;

exports[`Checkbox should render with onChecked 1`] = `
<div>
  <div
    class="checkbox"
    data-testid="checkbox"
  >
    <label />
    <input
      type="checkbox"
    />
  </div>
</div>
`;
